autorest/adal/token.go

```diff
@@ -258,6 +318,15 @@ func NewServicePrincipalToken(oauthConfig OAuthConfig, clientID string, secret s
 
 // NewServicePrincipalTokenFromCertificate create a ServicePrincipalToken from the supplied pkcs12 bytes.
 func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID string, certificate *x509.Certificate, privateKey *rsa.PrivateKey, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
```

Comment:
>certificate *x509.Certificate, privateKey *rsa.PrivateKey [](start = 87, length = 57)

`certificate` and `privateKey` can be nil without problem?
Since this func is `NewServicePrincipalTokenFromCertificate` I would guess at least cert should not be nil.

Reply:
Good catch, fixed.